airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/token.py

```diff
@@ -0,0 +1,57 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from itertools import cycle
+from typing import Any, List, Mapping
+
+from requests.auth import AuthBase
+
+
+class TokenAuthenticator(AuthBase):
```

Comment:
Can we make this class based on MultipleTokenAuthenticator? somthing like this:
```python
class TokenAuthenticator(MultipleTokenAuthenticator):
   def __init__(self, token: str, auth_method: str = "Bearer", auth_header: str = "Authorization"):
       super().__init__([token], auth_method, auth_header)
```

Reply:
@avida I think it will be good idea